app/components/InstantMessage/ImMain/Main.jsx

```diff
@@ -84,23 +103,15 @@ module.exports = React.createClass({
       if (!LoginStore.getUser()) {
         return;
       }
-      let channelIdToGo = params.backEndChannelId;
-      if (channelIdToGo === 'default' && localStorage[IMConstant.LOCALSTORAGE_CHANNEL]) {
-        // load from localStorage
-        channelIdToGo = localStorage[IMConstant.LOCALSTORAGE_CHANNEL];
-        // redirect
-        channelIdToGo && transition.redirect('/platform/im/talk/' + channelIdToGo);
-
-      } else {
-        if (!SocketStore.getSocket()) {
-          _channels = _init();
-          InitAction.init(_channels, LoginStore.getUser());
-        }
-        // validate channelIdToGo
-        if (isChannelIdValid(channelIdToGo)) {
-          ChannelAction.changeChannel(channelIdToGo, LoginStore.getUser());
-        }
+      if (UserStore.getUsersArray() && UserStore.getUsersArray().length === 0) {
+        // wait for the users array arrive
```

Comment:
To make it more reliable & expressive, you can add a variable `initialized` in UserStore, then use `UserStore.isInitialized()`